### PR TITLE
[DPE-5050] Add Loki support

### PR DESCRIPTION
**Pull request by welpaolo**
This PR introduces log forwarding from charm to Loki. 

This PR includes feature and integration tests.

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ resources:
   spark-history-server-image:
     type: oci-image
     description: OCI image for Spark History Server
-    upstream-source: ghcr.io/canonical/charmed-spark@sha256:90a1f26a33d7524b00b5b0692f03ed96919ea89ded571943d2b5aa7802389f87
+    upstream-source: ghcr.io/canonical/charmed-spark@sha256:4c4e6f9d394348a26ec66969898434f007467b2caee03b2d393ddffd14dc2ecf
```
Comment (theoctober19th) [inline metadata.yaml:23]: Can you please add the version + revision of this image as a comment?

File: tests/integration/test_charm_logs.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        status="active",
+        timeout=1000,
+    )
+
+
+@pytest.mark.abort_on_fail
+async def test_loki_integration(ops_test: OpsTest, charm_versions):
+    """Check that logs are forwarded to Loki.
+
+    Assert on the unit status before any relations/configurations take place.
```
Comment (theoctober19th) [inline tests/integration/test_charm_logs.py:126]: nit: update docstring

**Review by theoctober19th — APPROVED** [review]
LGTM. I've left minor non-blocking comments.

File: metadata.yaml (hunk 1 above)
Comment (deusebio) [reply metadata.yaml:23]: Yes, that a good idea!

**Review by deusebio — APPROVED** [review]
Looks very good! Nice and neat!

Just to make sure...have you checked that the logs that are collected are indeed from the history server? I didn't see any customization of the LogForwarder, I assume it is because those logs are direcntly into pebble (instead of actual files as in other platforms, like [here](https://github.com/canonical/kafka-k8s-operator/blob/main/src/charm.py#L83))

Just want to make sure that we are collecting proper logs, and not system logs only

**Pull request merged**